src/common/util.ts

```diff
@@ -87,12 +87,28 @@ export async function findWorkDirectory(
   const dirname = path.dirname(filePath)
   let patterns = [].concat(rootPatterns)
   try {
-    const dir = await findUp(patterns, {
-      cwd: dirname
-    });
+    for(const pattern of patterns) {
+      // attempt to match for directory patterns
+      let dirMatchPath = await findUp(pattern, {
+        cwd: dirname,
+        type: "directory",
+      })
+
+      if (!dirMatchPath) {
+        // if no directories were found, attempt to match file patterns
+        const fileMatchPath = await findUp(pattern, {
+          cwd: dirname,
+          type: "file",
+        })
 
-    if (dir && dir !== '/') {
-      return dir
+        if (fileMatchPath) {
```

Comment:
if dirMatchPath is `/path/to/xxx/.git` this is no need.

Reply:
what about file match, e.g. `composer.lock`? If passed pattern is a file then `dirMatchPath` will not contain any match because of the `type` filter.